src/gui/qgsmaptooldigitizefeature.cpp

```diff
@@ -92,6 +92,40 @@ void QgsMapToolDigitizeFeature::setCheckGeometryType( bool checkGeometryType )
   mCheckGeometryType = checkGeometryType;
 }
 
+void QgsMapToolDigitizeFeature::keyPressEvent( QKeyEvent *e )
+{
+  QgsVectorLayer *vlayer = qobject_cast<QgsVectorLayer *>( mLayer );
+  if ( !vlayer )
+    //if no given layer take the current from canvas
+    vlayer = currentVectorLayer();
+
+  if ( !vlayer )
+  {
+    notifyNotVectorLayer();
+    return;
+  }
+
+  QgsVectorDataProvider *provider = vlayer->dataProvider();
+
+  if ( !( provider->capabilities() & QgsVectorDataProvider::AddFeatures ) )
+  {
+    emit messageEmitted( tr( "The data provider for this layer does not support the addition of features." ), Qgis::Warning );
+    return;
+  }
+
+  if ( !vlayer->isEditable() )
+  {
+    notifyNotEditableLayer();
+    return;
+  }
+
+  if ( e && e->key() == Qt::Key_C && mode() == CaptureLine && vlayer->geometryType() == QgsWkbTypes::LineGeometry && mCheckGeometryType )
```

Comment:
shouldn't this include support for CapturePolygon mode? I realise that a right click will auto-close a polygon, but for consistency across the tools I think 'c' should also close and end polygon digitization.

Reply:
I hesitated, but indeed, it is consistent like that.